### PR TITLE
Hosting Metrics: Clean up old Site Logs code

### PR DESCRIPTION
**Pull request by danielbachhuber**
Fixes https://github.com/Automattic/dotcom-forge/issues/3393

## Proposed Changes

* Removes legacy Site logs code.
* Redirects `/site-logs*` to `/site-monitoring*`
* Updates 'Site logs' to 'Site monitoring' in the ellipsis menu.

CleanShot 2023-08-28 at 15 48 09@2x

## Testing Instructions

1. Navigate to `/site-logs` and verify you're redirected to `/site-monitoring` as expected.
2. Navigate to `/site-logs/<site>` and verify you're redirected to `/site-monitoring/<site>` as expected.
3. Verify the 'Site monitoring' ellipsis menu item appears and works as expected.

### CODE REVIEW

File: client/my-sites/site-monitoring/index.ts [hunk 1]
```diff
@@ -15,4 +15,16 @@ export default function () {
 		makeLayout,
 		clientRender
 	);
+
+	// Legacy redirect for Site Logs.
+	const redirectSiteLogsToMonitoring: PageJS.Callback = ( context ) => {
+		if ( context.params?.siteId ) {
+			context.page.replace( `/site-monitoring/${ context.params.siteId }` );
```
Comment (p-jackson) [inline client/my-sites/site-monitoring/index.ts:22]: This drops query parameters, which means if a user has bookmarked a specific logging page, that bookmark now takes them to the overall monitoring view. I would expect it to take them to the same logging view as before.

I can see a case for not worrying about it and perhaps it even being a good way to advertise the new monitoring features. On the other hand we know users are using the logs and it seems reasonable to expect their bookmarks would keep workign.

**Review by p-jackson — APPROVED** [review]
LGTM

In a follow up we can also remove the "Site logs" item from the sidebar.

The first place is in `fallback-menu.js` which provides menu items before the unified-nav is loaded from the server.
https://github.com/Automattic/wp-calypso/blob/9f0f2f0e7db56aa955ec7b4868ab944b8c56b943/client/my-sites/sidebar/static-data/fallback-menu.js#L526

This is just the fallback which gives the user a temporary menu before the official one gets loaded. To demonstrate this you can search for `admin-menu` in the network tools, block this network request, and then refresh that page. You'll see "Site logs" is in the Tools menu.

The other place to remove the menu item is from Jetpack.
https://github.com/Automattic/jetpack/blob/583e9ca95b90f36fee3c22456c892ea9c56223d9/projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php#L401

**Comment by danielbachhuber** [comment]
> This is just the fallback which gives the user a temporary menu before the official one gets loaded. To demonstrate this you can search for `admin-menu` in the network tools, block this network request, and then refresh that page. You'll see "Site logs" is in the Tools menu.

Good catch, updated with d69e8b1b24

> The other place to remove the menu item is from Jetpack.

We have a PR going for this https://github.com/Automattic/jetpack/pull/32458

---

Thanks @p-jackson !

File: client/my-sites/site-monitoring/index.ts (hunk 1 above)
Comment (danielbachhuber) [reply client/my-sites/site-monitoring/index.ts:22]: Thanks for flagging. I'm not so concerned about it, but it's a good point to consider.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8824181).

**Pull request merged**